### ISSUE TITLE
manual linux installation; step 6; make errors

### ISSUE THREAD

**Issue by mushrushu** [intel/dptf#21]
gcc --version
gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)

cd ESIF/Products/ESIF_UF/Linux
make
++++ 
++++ ESIF_UF x64 Release Linux GMIN=1
++++ 
gcc -c -Wno-multichar -Werror -fPIE -fPIC -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -fstack-protector -m64  -I../../ESIF_UF/Sources -I../../ESIF_CM/Sources -I../../ESIF_LIB/Sources -I../../../../Common -DESIF_ATTR_USER  -DESIF_ATTR_DAEMON -DESIF_FEAT_OPT_COMPRESS -DESIF_ATTR_OS_LINUX -DESIF_ATTR_OS_LINUX_HAVE_READLINE  -std=gnu89 -DESIF_FEAT_OPT_ACTION_SYSFS -o ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.o ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c
In file included from ../../../../Common/esif_ccb_string.h:72:0,
                 from ../../../../Common/esif_ccb_library_lin_user.h:28,
                 from ../../../../Common/esif_ccb_library.h:28,
                 from ../../ESIF_CM/Sources/esif.h:207,
                 from ../../ESIF_UF/Sources/esif_uf.h:22,
                 from ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:23:
../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c: In function ‘ResetThermalZonePolicy’:
../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:2103:52: error: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 237 [-Werror=format-truncation=]
    esif_ccb_sprintf(MAX_SYSFS_PATH, cur_node_name, "%s/%s", SYSFS_THERMAL,namelist[n]->d_name);
                                                    ^
../../../../Common/esif_ccb_string_lin_user.h:67:65: note: in definition of macro ‘esif_ccb_sprintf’
 #define esif_ccb_sprintf(siz, str, fmt, ...) snprintf(str, siz, fmt, ##__VA_ARGS__)
                                                                 ^~~
In file included from /usr/include/stdio.h:862:0,
                 from ../../../../Common/esif_ccb.h:58,
                 from ../../../../Common/esif_sdk.h:56,
                 from ../../ESIF_CM/Sources/esif.h:56,
                 from ../../ESIF_UF/Sources/esif_uf.h:22,
                 from ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:23:
/usr/include/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 20 and 275 bytes into a destination of size 256
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../Common/esif_ccb_string.h:72:0,
                 from ../../../../Common/esif_ccb_library_lin_user.h:28,
                 from ../../../../Common/esif_ccb_library.h:28,
                 from ../../ESIF_CM/Sources/esif.h:207,
                 from ../../ESIF_UF/Sources/esif_uf.h:22,
                 from ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:23:
../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c: In function ‘SetThermalZonePolicy’:
../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:2057:52: error: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 237 [-Werror=format-truncation=]
    esif_ccb_sprintf(MAX_SYSFS_PATH, cur_node_name, "%s/%s", SYSFS_THERMAL,namelist[n]->d_name);
                                                    ^
../../../../Common/esif_ccb_string_lin_user.h:67:65: note: in definition of macro ‘esif_ccb_sprintf’
 #define esif_ccb_sprintf(siz, str, fmt, ...) snprintf(str, siz, fmt, ##__VA_ARGS__)
                                                                 ^~~
In file included from /usr/include/stdio.h:862:0,
                 from ../../../../Common/esif_ccb.h:58,
                 from ../../../../Common/esif_sdk.h:56,
                 from ../../ESIF_CM/Sources/esif.h:56,
                 from ../../ESIF_UF/Sources/esif_uf.h:22,
                 from ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.c:23:
/usr/include/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 20 and 275 bytes into a destination of size 256
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:181: ../../ESIF_UF/Sources/lin/esif_uf_action_sysfs_os_lin.o] Error 1

**Comment by JeffLabonte** [comment]
Same issue here  on Ubuntu 18.04

```
gcc (Ubuntu 7.3.0-16ubuntu3) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOS
```E

**Comment by bbian** [comment]
Commit f6983da26683086b3431f8b7c16528e7474ede25 should fix compilation on Ubuntu 18.04.
